components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -5637,8 +5637,14 @@ public Map<String, Object> searchPaginatedAPIs(String searchQuery, String organi
                 List<Object> apiList = new ArrayList<>();
                 for (DevPortalAPIInfo devPortalAPIInfo : list) {
                     API mappedAPI = APIMapper.INSTANCE.toApi(devPortalAPIInfo);
-                    mappedAPI.setRating(APIUtil.getAverageRating(mappedAPI.getUuid()));
-                    apiList.add(mappedAPI);
+                    try {
+                        mappedAPI.setRating(APIUtil.getAverageRating(mappedAPI.getUuid()));
+                        apiList.add(mappedAPI);
+                    } catch (APIManagementException e) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Getting Average rating failed for API: " + mappedAPI.getUuid());
+                        }
+                    }
```

Comment:
The `searchPaginatedAPIs()` method declares that it throws `APIManagementException`. So catching it here and not rethrowing means the flow is now altered. i.e., a previous flow where `searchPaginatedAPIs()` terminated with an exception will now return as if nothing happened. Is that behavioural change intended?

Reply:
Yes. 
The intention is to do `apiList.add(mappedAPI);` to only `mappedAPI`s (in the `for` loop) that did not get `APIManagementException` when calling `APIUtil.getAverageRating()`.

Earlier it threw `APIManagementException` and considered as an error. Here it will skip the erroneous iteration of the `for ` loop and add the rest to the list.